src/routes/Profile/FreeTimeActivities/modals/EditPatevyysModal.tsx

```diff
@@ -0,0 +1,288 @@
+        render={({ field: { onChange, value } }) => (
+          <OsaamisSuosittelija
+            description={debouncedDescription}
+            onChange={onChange}
+            value={value}
+            sourceType="PATEVYYS"
+          />
+        )}
+      />
+    </>
+  );
+};
+
+export const EditPatevyysModal = ({ isOpen, onClose, toimintoId: id, patevyysId }: EditPatevyysProps) => {
+  const {
+    t,
+    i18n: { language },
+  } = useTranslation();
+
+  if (!id) {
+    onClose();
+  }
+
+  const [toiminto, setToiminto] = React.useState<components['schemas']['ToimintoDto'] | undefined>(undefined);
+  const formId = React.useId();
+  const [step, setStep] = React.useState(0);
+  const stepComponents = [MainStep, OsaaminenStep];
+  const StepComponent = stepComponents[step];
+
+  const nextStep = () => {
+    setStep(step < stepComponents.length - 1 ? step + 1 : step);
+  };
+  const previousStep = () => {
+    setStep(step > 0 ? step - 1 : step);
+  };
+  const isLastStep = step === stepComponents.length - 1;
+  const isFirstStep = step === 0;
+
+  const methods = useForm<PatevyysForm>({
+    mode: 'onChange',
+    resolver: zodResolver(
+      z
+        .object({
+          id: z.string(),
+          nimi: z.string().min(1),
+          alkuPvm: z.string().date(),
+          loppuPvm: z.string().date().optional().or(z.literal('')),
+          osaamiset: z.array(
+            z.object({
+              id: z.string().min(1),
+            }),
+          ),
+        })
+        .refine((data) => !data.loppuPvm || data.alkuPvm <= data.loppuPvm),
+    ),
+    defaultValues: async () => {
+      const { data: osaamiset } = await client.GET('/api/profiili/osaamiset');
+      const { data: toimintoData } = await client.GET('/api/profiili/vapaa-ajan-toiminnot/{id}', {
+        params: {
+          path: { id },
+        },
+      });
+
+      setToiminto(toimintoData);
+      const patevyys = toimintoData?.patevyydet?.find((p) => p.id === patevyysId);
+
+      return {
+        id: patevyys?.id ?? '',
+        nimi: patevyys?.nimi?.[language] ?? '',
+        alkuPvm: patevyys?.alkuPvm ?? '',
+        loppuPvm: patevyys?.loppuPvm ?? '',
+        osaamiset:
+          patevyys?.osaamiset?.map((osaaminenId) => ({
+            id: osaaminenId,
+            nimi: osaamiset?.find((o) => o.osaaminen?.uri === osaaminenId)?.osaaminen?.nimi?.[language] ?? '',
+          })) ?? [],
+      };
+    },
+  });
+  const trigger = methods.trigger;
+  const { isLoading, isValid } = useFormState({
+    control: methods.control,
+  });
+
+  const onSubmit: FormSubmitHandler<PatevyysForm> = async ({ data }: { data: PatevyysForm }) => {
+    if (!toiminto) {
+      return;
+    }
+
+    const params = {
+      params: {
+        path: {
+          id,
+        },
+      },
+      body: {
+        id: toiminto.id,
+        nimi: toiminto.nimi,
+        patevyydet:
+          toiminto?.patevyydet?.map((p) => {
+            if (p.id === patevyysId) {
```

Comment:
Since `patevyys` doesn't have it's own update endpoint, the whole "vapaa-ajan toiminto" needs to be updated. Maybe this will get changed in OPHJOD-457